src/planner/ddl_plan.go

```diff
@@ -66,162 +65,215 @@ func NewDDLPlan(log *xlog.Log, database string, query string, node *sqlparser.DD
+func (p *DDLPlan) commonImpl() error {
+	oldNode := p.node
+	oldTable := oldNode.Table.Name.String()
+	database := p.database
+	if !oldNode.Table.Qualifier.IsEmpty() {
+		database = oldNode.Table.Qualifier.String()
+	}
+	if err := p.checkUnsupportedOperations(database, oldTable); err != nil {
+		return err
+	}
+
+	segments, err := p.router.Lookup(database, oldTable, nil, nil)
+	if err != nil {
+		return err
+	}
+	for _, segment := range segments {
+		// Rewrite ddl ast, replace oldTable to segment table(new table) and format a new query
+		newNode := *oldNode
+		newTable := segment.Table
+		buf := sqlparser.NewTrackedBuffer(nil)
+		newNode.Table = sqlparser.TableName{
+			Name:      sqlparser.NewTableIdent(newTable),
+			Qualifier: sqlparser.NewTableIdent(database),
 		}
-		for _, segment := range segments {
-			var query string
-
-			orgSegTable := segment.Table
-			var rawQuery string
-			var re *regexp.Regexp
-			var segTable string
-			if node.Table.Qualifier.IsEmpty() {
-				segTable = fmt.Sprintf("`%s`.`%s`", database, orgSegTable)
-				rawQuery = strings.Replace(p.RawQuery, "`", "", -1)
-				// \b: https://www.regular-expressions.info/wordboundaries.html
-				re, _ = regexp.Compile(fmt.Sprintf(`\b(%s)\b`, table))
-			} else {
-				segTable = fmt.Sprintf("`%s`.`%s`", database, orgSegTable)
-				newTable := fmt.Sprintf("%s.%s", database, table)
-				rawQuery = strings.Replace(p.RawQuery, "`", "", -1)
-				re, _ = regexp.Compile(fmt.Sprintf(`\b(%s)\b`, newTable))
-			}
+		newNode.NewName = newNode.Table
+		newNode.Format(buf)
+		newQuery := buf.String()
 
-			// avoid the name of the column is the same as the table name, eg, issues/438
-			// just replace the first place.
-			var count = 0
-			var occurrence = 1
-			query = re.ReplaceAllStringFunc(rawQuery, func(m string) string {
-				count = count + 1
-				if count == occurrence {
-					return segTable
-				}
-				return m
-			})
-
-			if node.Action == sqlparser.RenameStr {
-				var segQuery string
-				var segToTable string
-				var re *regexp.Regexp
-				pos := strings.Index(query, segTable)
-				pos += len(segTable)
-
-				if !node.NewName.Qualifier.IsEmpty() {
-					toDatabase := node.NewName.Qualifier.String()
-					if toDatabase != database {
-						err := fmt.Sprintf("unsupported: Database is not equal[%s:%s]", database, toDatabase)
-						return errors.New(err)
-					}
-				}
+		tuple := xcontext.QueryTuple{
+			Query:   newQuery,
+			Backend: segment.Backend,
+			Range:   segment.Range.String(),
+		}
+		p.Querys = append(p.Querys, tuple)
+	}
+	return nil
+}
 
-				toTable := node.NewName.Name.String()
-				// just to the shardtable, the suffix with "_0001" is valid
-				if shardKey != "" {
-					splits := strings.SplitN(orgSegTable, "_", -1)
-					suffix := splits[len(splits)-1]
-					segToTable = toTable + "_" + suffix
-				} else {
-					segToTable = toTable
-				}
+// dropTblImpl used to build distributed querys for drop table tbl1, tbl2 ...
```

Comment:
`drop tbl1`, not `drop table tbl1, tbl2`

Reply:
Done!